js/web/lib/wasm/jsep/webgpu/ops/softmax.ts

```diff
@@ -52,18 +50,24 @@ const createSoftmaxProgramInfo = (input: TensorView, attributes: SoftmaxAttribut
 
     return name;
   };
-
+  const x = inputVariable('x', input.dataType, input.dims, components);
+  const output = outputVariable('result', input.dataType, input.dims, components);
+  const enableInputShapesUniforms = enableShapesUniforms(input.dims.length);
+  const programUniforms: ProgramUniform[] = [{type: 'uint32', data: packedCols}];
+  if (enableInputShapesUniforms) {
+    programUniforms.push(...createTensorShapeVariables(input.dims), ...createTensorShapeVariables(input.dims));
+  }
```

Comment:
Since you are depending on `type` instead of `rank`. L57~L59 is not needed anymore?

Reply:
done